packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx

```diff
@@ -28,104 +40,71 @@ const _testTags: ITag[] = [
 
-  public render() {
-    return (
-      <div className={rootClass}>
-        <Checkbox
-          styles={checkboxStyles}
-          label="Disable Tag Picker"
-          checked={this.state.isPickerDisabled}
-          onChange={this._onDisabledButtonClick}
-        />
-        Filter items in suggestions: This picker will filter added items from the search suggestions.
-        <TagPicker
-          removeButtonAriaLabel="Remove"
-          onResolveSuggestions={this._onFilterChanged}
-          getTextFromItem={this._getTextFromItem}
-          pickerSuggestionsProps={{
-            suggestionsHeaderText: 'Suggested Tags',
-            noResultsFoundText: 'No Color Tags Found',
-          }}
-          itemLimit={2}
-          disabled={this.state.isPickerDisabled}
-          inputProps={{
-            onBlur: (ev: React.FocusEvent<HTMLInputElement>) => console.log('onBlur called'),
-            onFocus: (ev: React.FocusEvent<HTMLInputElement>) => console.log('onFocus called'),
-            'aria-label': 'Tag Picker',
-          }}
-        />
-        <br />
-        Filter items on selected: This picker will show already-added suggestions but will not add duplicate tags.
-        <TagPicker
-          removeButtonAriaLabel="Remove"
-          componentRef={this._picker}
-          onResolveSuggestions={this._onFilterChangedNoFilter}
-          onItemSelected={this._onItemSelected}
-          getTextFromItem={this._getTextFromItem}
-          pickerSuggestionsProps={{
-            suggestionsHeaderText: 'Suggested Tags',
-            noResultsFoundText: 'No Color Tags Found',
-          }}
-          itemLimit={2}
-          disabled={this.state.isPickerDisabled}
-          inputProps={{
-            onBlur: (ev: React.FocusEvent<HTMLInputElement>) => console.log('onBlur called'),
-            onFocus: (ev: React.FocusEvent<HTMLInputElement>) => console.log('onFocus called'),
-            'aria-label': 'Tag Picker',
-          }}
-        />
-      </div>
-    );
-  }
-
-  private _getTextFromItem(item: ITag): string {
-    return item.name;
-  }
-
-  private _onDisabledButtonClick = (): void => {
-    this.setState({
-      isPickerDisabled: !this.state.isPickerDisabled,
-    });
-  };
+  const listContainsDocument = useConstCallback((tag: ITag, tagList?: ITag[]) => {
+    if (!tagList || !tagList.length || tagList.length === 0) {
+      return false;
+    }
+    return tagList.filter(compareTag => compareTag.key === tag.key).length > 0;
+  });
 
-  private _onFilterChanged = (filterText: string, tagList: ITag[]): ITag[] => {
+  const onFilterChanged = useConstCallback((filterText: string, tagList: ITag[]): ITag[] => {
     return filterText
-      ? _testTags
+      ? testTags
           .filter(tag => tag.name.toLowerCase().indexOf(filterText.toLowerCase()) === 0)
-          .filter(tag => !this._listContainsDocument(tag, tagList))
+          .filter(tag => !listContainsDocument(tag, tagList))
```

Comment:
Can combine this: `.filter(tag => tag.name.toLowerCase().indexOf(filterText.toLowerCase()) === 0 && !listContainsDocument(tag, tagList))`